dtfs-central-api/src/v1/controllers/utilisation-report-service/get-utilisation-report-reconciliation-details-by-id.controller/helpers/get-keying-sheet-for-report-id.test.ts

```diff
@@ -272,6 +250,136 @@ describe('getKeyingSheetForReportId', () => {
     expect(keyingSheet[0].feePayments[2].dateReceived).toEqual(new Date('2023'));
   });
 
+  it('sets the keying sheet row fee payment date to null and amount to zero when the join table paymentAmountUsedForFeeRecord is null', async () => {
+    // Arrange
+    const firstPayment = PaymentEntityMockBuilder.forCurrency('GBP').withId(11).withDateReceived(new Date('2021')).build();
+    const secondPayment = PaymentEntityMockBuilder.forCurrency('USD').withDateReceived(new Date('2022')).withId(22).build();
+
+    const feeRecord = FeeRecordEntityMockBuilder.forReport(aUtilisationReport()).withStatus('READY_TO_KEY').withId(12).build();
+
+    const joinTableEntities = [
+      aFeeRecordPaymentJoinTableEntityWith({ feeRecord, payment: firstPayment, paymentAmountUsedForFeeRecord: 1000 }),
+      aFeeRecordPaymentJoinTableEntityWith({ feeRecord, payment: secondPayment, paymentAmountUsedForFeeRecord: null }),
+    ];
+
+    const reportId = 1;
+    when(mockFind)
+      .calledWith(FeeRecordPaymentJoinTableEntity, {
+        where: {
+          feeRecord: {
+            report: { id: reportId },
+            status: In<FeeRecordStatus>(['READY_TO_KEY', 'RECONCILED']),
+          },
+        },
+        relations: {
+          feeRecord: true,
+          payment: true,
+        },
+      })
+      .mockResolvedValue(joinTableEntities);
+
+    // Act
+    const keyingSheet = await getKeyingSheetForReportId(reportId, [feeRecord]);
+
+    // Assert
+    expect(keyingSheet).toHaveLength(1);
+    expect(keyingSheet[0].feePayments).toHaveLength(2);
+    expect(keyingSheet[0].feePayments[0].currency).toBe('GBP');
+    expect(keyingSheet[0].feePayments[0].amount).toBe(1000);
+    expect(keyingSheet[0].feePayments[0].dateReceived).toEqual(new Date('2021'));
+    expect(keyingSheet[0].feePayments[1].currency).toBe('USD');
+    expect(keyingSheet[0].feePayments[1].amount).toBe(0);
+    expect(keyingSheet[0].feePayments[1].dateReceived).toBeNull();
+  });
```

Comment:
Yes it would. I suppose we only want a zero fee payment to appear in the case when there are no other fee payments, so I'll update the logic and test to do that